src/lib.zig

```diff
@@ -563,154 +563,8 @@ pub fn chunkCount(comptime T: type) usize {
     }
 }
 
-const chunk = [BYTES_PER_CHUNK]u8;
-const zero_chunk: chunk = [_]u8{0} ** BYTES_PER_CHUNK;
-
-fn pack(comptime T: type, values: T, l: *ArrayList(u8)) ![]chunk {
```

Comment:
I know @g11tech told you to move stuff to utils, but here is not the place to do that.  This makes reviewing this PR much more work. Please move stuff from utils back here. If Gajinder can not live with this being here, which is where I think it belongs, let's create another PR for this, and only keep in this PR the changes that only do what the PR is indended to do.